pkg/agent/route/route_windows.go

```diff
@@ -280,28 +280,43 @@ func (c *Client) addServiceCIDRRoute(serviceCIDR *net.IPNet) error {
 	c.serviceRoutes.Store(serviceIPv4CIDRKey, route)
 
 	// Collect stale routes.
-	var staleRoutes []*util.Route
-	// If current destination CIDR is not nil, the route with current destination CIDR should be uninstalled since
-	// a new route with a newly calculated destination CIDR has been installed.
+	staleRoutes := make(map[string]*util.Route)
 	if serviceCIDRRouteExists {
-		staleRoutes = append(staleRoutes, oldServiceCIDRRoute.(*util.Route))
+		// If current destination CIDR is not nil, the route with current destination CIDR should be uninstalled since
+		// a new route with a newly calculated destination CIDR has been installed.
+		rt := oldServiceCIDRRoute.(*util.Route)
+		staleRoutes[rt.String()] = rt
+	} else {
+		// If current destination CIDR is nil, which means that Antrea Agent has just started, then all existing routes
+		// whose destination CIDR contains the first ClusterIP should be uninstalled, except the newly installed route.
+		// Note that, there may be multiple stale routes prior to this commit. When upgrading, all stale routes will be
+		// collected. After this commit, there will be only one stale route after Antrea Agent started.
+		routes, err := c.listIPRoutesOnGW()
+		if err != nil {
+			return fmt.Errorf("error listing ip routes: %w", err)
+		}
+		for _, rt := range routes {
+			if rt.GatewayAddress.Equal(gw) && !rt.DestinationSubnet.IP.Equal(serviceCIDR.IP) && rt.DestinationSubnet.Contains(serviceCIDR.IP) {
```

Comment:
Why not use the similar logic as line 310 to line 312, like `iputil.IsSubCIDR(rt.DestinationSubnet, serviceCIDR)`?

Reply:
This is used to consistent with Linux code. Besides, the logic is used to find the existing routes whose destinations contains the first ClusterIP (serviceCIDR.IP), and it is only called when syncing the first ClusterIP.